fhir-server-test/src/test/java/com/ibm/fhir/server/test/operation/ReindexOperationPhase2Test.java

```diff
@@ -0,0 +1,297 @@
+
+        Parameters.Builder builder = Parameters.builder();
+        builder.id(UUID.randomUUID().toString());
+        builder.parameter(parameters);
+        Parameters ps = builder.build();
+
+        Entity<Parameters> entity = Entity.entity(ps, FHIRMediaType.APPLICATION_FHIR_JSON);
+
+        Response r = getWebTarget()
+                .path("/Patient/REIN-DEX-TEST-1/$reindex")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .header("X-FHIR-TENANT-ID", "default")
+                .header("X-FHIR-DSID", "default")
+                .post(entity, Response.class);
+
+        assertEquals(r.getStatus(), Status.OK.getStatusCode());
+    }
+
+    /**
+     * reindex a specific Basic/id resource
+     */
+    private void reindex(String id) {
+        List<Parameter> parameters = new ArrayList<>();
+        parameters.add(Parameter.builder()
+            .name(string("resourceLogicalId"))
+            .value(string("Basic/" + id))
+            .build());
+
+        Parameters.Builder builder = Parameters.builder();
+        builder.id(UUID.randomUUID().toString());
+        builder.parameter(parameters);
+        Parameters ps = builder.build();
+
+        Entity<Parameters> entity = Entity.entity(ps, FHIRMediaType.APPLICATION_FHIR_JSON);
+
+        Response response = getWebTarget()
+                .path("/$reindex")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .header("X-FHIR-TENANT-ID", "default")
+                .header("X-FHIR-DSID", "default")
+                .post(entity, Response.class);
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        OperationOutcome oo = response.readEntity(OperationOutcome.class);
+        assertEquals(oo.getIssue().get(0).getDiagnostics().getValue(), "Processed Basic/" + id);
+    }
+
+    /**
+     * checks the basic resource is found
+     * @param code
+     * @param id
+     * @param value
+     */
+    private void verifyFoundBySearch(String code, String id, String value) {
+        // Search and Confirm the search using the 'code' does returns a resource.
+        Response response = getWebTarget().path("/Basic")
+                .queryParam(code, value)
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .header("X-FHIR-TENANT-ID", "default")
+                .header("X-FHIR-DSID", "default")
+                .get(Response.class);
+        assertEquals(response.getStatus(), Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+        assertFalse(bundle.getEntry().isEmpty());
+        assertEquals(bundle.getEntry().get(0).getId(), id);
+    }
+
+    @Test(groups = {"reindex"}, dependsOnMethods = {})
+    public void testReindex_String_Phase2() throws IOException, FHIRParserException, FHIRPathException {
+        if (runIt) {
+            reindex("REINDEX-STRING");
+            verifyFoundBySearch("reindex-string", "REINDEX-STRING", "");
```

Comment:
Shouldn't the value be "testString" to match BasicString.json?

Reply:
Fixed